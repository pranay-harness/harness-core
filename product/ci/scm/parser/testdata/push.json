{
    "ref": "refs/heads/master",
    "before": "e286f86f4db84e682b50007772a28a9d5e9b51d0",
    "after": "3c377597b64e441cae5378b660578fa00983aaba",
    "repository": {
      "id": 288719292,
      "node_id": "MDEwOlJlcG9zaXRvcnkyODg3MTkyOTI=",
      "name": "test-webhook",
      "full_name": "wings-software/test-webhook",
      "private": true,
      "owner": {
        "name": "wings-software",
        "email": null,
        "login": "wings-software",
        "id": 18273000,
        "node_id": "MDEyOk9yZ2FuaXphdGlvbjE4MjczMDAw",
        "avatar_url": "https://avatars1.githubusercontent.com/u/18273000?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/wings-software",
        "html_url": "https://github.com/wings-software",
        "followers_url": "https://api.github.com/users/wings-software/followers",
        "following_url": "https://api.github.com/users/wings-software/following{/other_user}",
        "gists_url": "https://api.github.com/users/wings-software/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/wings-software/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/wings-software/subscriptions",
        "organizations_url": "https://api.github.com/users/wings-software/orgs",
        "repos_url": "https://api.github.com/users/wings-software/repos",
        "events_url": "https://api.github.com/users/wings-software/events{/privacy}",
        "received_events_url": "https://api.github.com/users/wings-software/received_events",
        "type": "Organization",
        "site_admin": false
      },
      "html_url": "https://github.com/wings-software/test-webhook",
      "description": null,
      "fork": false,
      "url": "https://github.com/wings-software/test-webhook",
      "forks_url": "https://api.github.com/repos/wings-software/test-webhook/forks",
      "keys_url": "https://api.github.com/repos/wings-software/test-webhook/keys{/key_id}",
      "collaborators_url": "https://api.github.com/repos/wings-software/test-webhook/collaborators{/collaborator}",
      "teams_url": "https://api.github.com/repos/wings-software/test-webhook/teams",
      "hooks_url": "https://api.github.com/repos/wings-software/test-webhook/hooks",
      "issue_events_url": "https://api.github.com/repos/wings-software/test-webhook/issues/events{/number}",
      "events_url": "https://api.github.com/repos/wings-software/test-webhook/events",
      "assignees_url": "https://api.github.com/repos/wings-software/test-webhook/assignees{/user}",
      "branches_url": "https://api.github.com/repos/wings-software/test-webhook/branches{/branch}",
      "tags_url": "https://api.github.com/repos/wings-software/test-webhook/tags",
      "blobs_url": "https://api.github.com/repos/wings-software/test-webhook/git/blobs{/sha}",
      "git_tags_url": "https://api.github.com/repos/wings-software/test-webhook/git/tags{/sha}",
      "git_refs_url": "https://api.github.com/repos/wings-software/test-webhook/git/refs{/sha}",
      "trees_url": "https://api.github.com/repos/wings-software/test-webhook/git/trees{/sha}",
      "statuses_url": "https://api.github.com/repos/wings-software/test-webhook/statuses/{sha}",
      "languages_url": "https://api.github.com/repos/wings-software/test-webhook/languages",
      "stargazers_url": "https://api.github.com/repos/wings-software/test-webhook/stargazers",
      "contributors_url": "https://api.github.com/repos/wings-software/test-webhook/contributors",
      "subscribers_url": "https://api.github.com/repos/wings-software/test-webhook/subscribers",
      "subscription_url": "https://api.github.com/repos/wings-software/test-webhook/subscription",
      "commits_url": "https://api.github.com/repos/wings-software/test-webhook/commits{/sha}",
      "git_commits_url": "https://api.github.com/repos/wings-software/test-webhook/git/commits{/sha}",
      "comments_url": "https://api.github.com/repos/wings-software/test-webhook/comments{/number}",
      "issue_comment_url": "https://api.github.com/repos/wings-software/test-webhook/issues/comments{/number}",
      "contents_url": "https://api.github.com/repos/wings-software/test-webhook/contents/{+path}",
      "compare_url": "https://api.github.com/repos/wings-software/test-webhook/compare/{base}...{head}",
      "merges_url": "https://api.github.com/repos/wings-software/test-webhook/merges",
      "archive_url": "https://api.github.com/repos/wings-software/test-webhook/{archive_format}{/ref}",
      "downloads_url": "https://api.github.com/repos/wings-software/test-webhook/downloads",
      "issues_url": "https://api.github.com/repos/wings-software/test-webhook/issues{/number}",
      "pulls_url": "https://api.github.com/repos/wings-software/test-webhook/pulls{/number}",
      "milestones_url": "https://api.github.com/repos/wings-software/test-webhook/milestones{/number}",
      "notifications_url": "https://api.github.com/repos/wings-software/test-webhook/notifications{?since,all,participating}",
      "labels_url": "https://api.github.com/repos/wings-software/test-webhook/labels{/name}",
      "releases_url": "https://api.github.com/repos/wings-software/test-webhook/releases{/id}",
      "deployments_url": "https://api.github.com/repos/wings-software/test-webhook/deployments",
      "created_at": 1597838277,
      "updated_at": "2020-08-19T13:19:26Z",
      "pushed_at": 1598340602,
      "git_url": "git://github.com/wings-software/test-webhook.git",
      "ssh_url": "git@github.com:wings-software/test-webhook.git",
      "clone_url": "https://github.com/wings-software/test-webhook.git",
      "svn_url": "https://github.com/wings-software/test-webhook",
      "homepage": null,
      "size": 1,
      "stargazers_count": 0,
      "watchers_count": 0,
      "language": null,
      "has_issues": true,
      "has_projects": true,
      "has_downloads": true,
      "has_wiki": true,
      "has_pages": false,
      "forks_count": 0,
      "mirror_url": null,
      "archived": false,
      "disabled": false,
      "open_issues_count": 1,
      "license": null,
      "forks": 0,
      "open_issues": 1,
      "watchers": 0,
      "default_branch": "master",
      "stargazers": 0,
      "master_branch": "master",
      "organization": "wings-software"
    },
    "pusher": {
      "name": "shubham149",
      "email": "shubham.agrawal@harness.io"
    },
    "organization": {
      "login": "wings-software",
      "id": 18273000,
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjE4MjczMDAw",
      "url": "https://api.github.com/orgs/wings-software",
      "repos_url": "https://api.github.com/orgs/wings-software/repos",
      "events_url": "https://api.github.com/orgs/wings-software/events",
      "hooks_url": "https://api.github.com/orgs/wings-software/hooks",
      "issues_url": "https://api.github.com/orgs/wings-software/issues",
      "members_url": "https://api.github.com/orgs/wings-software/members{/member}",
      "public_members_url": "https://api.github.com/orgs/wings-software/public_members{/member}",
      "avatar_url": "https://avatars1.githubusercontent.com/u/18273000?v=4",
      "description": ""
    },
    "sender": {
      "login": "shubham149",
      "id": 61136843,
      "node_id": "MDQ6VXNlcjYxMTM2ODQz",
      "avatar_url": "https://avatars1.githubusercontent.com/u/61136843?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/shubham149",
      "html_url": "https://github.com/shubham149",
      "followers_url": "https://api.github.com/users/shubham149/followers",
      "following_url": "https://api.github.com/users/shubham149/following{/other_user}",
      "gists_url": "https://api.github.com/users/shubham149/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/shubham149/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/shubham149/subscriptions",
      "organizations_url": "https://api.github.com/users/shubham149/orgs",
      "repos_url": "https://api.github.com/users/shubham149/repos",
      "events_url": "https://api.github.com/users/shubham149/events{/privacy}",
      "received_events_url": "https://api.github.com/users/shubham149/received_events",
      "type": "User",
      "site_admin": false
    },
    "created": false,
    "deleted": false,
    "forced": false,
    "base_ref": null,
    "compare": "https://github.com/wings-software/test-webhook/compare/e286f86f4db8...3c377597b64e",
    "commits": [
      {
        "id": "3c377597b64e441cae5378b660578fa00983aaba",
        "tree_id": "a3d993b0df4984ead559a30b56f7bae0947fd01f",
        "distinct": true,
        "message": "Update README.md",
        "timestamp": "2020-08-25T13:00:01+05:30",
        "url": "https://github.com/wings-software/test-webhook/commit/3c377597b64e441cae5378b660578fa00983aaba",
        "author": {
          "name": "Shubham Agrawal",
          "email": "shubham.agrawal@harness.io",
          "username": "shubham149"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "username": "web-flow"
        },
        "added": [
  
        ],
        "removed": [
  
        ],
        "modified": [
          "README.md"
        ]
      }
    ],
    "head_commit": {
      "id": "3c377597b64e441cae5378b660578fa00983aaba",
      "tree_id": "a3d993b0df4984ead559a30b56f7bae0947fd01f",
      "distinct": true,
      "message": "Update README.md",
      "timestamp": "2020-08-25T13:00:01+05:30",
      "url": "https://github.com/wings-software/test-webhook/commit/3c377597b64e441cae5378b660578fa00983aaba",
      "author": {
        "name": "Shubham Agrawal",
        "email": "shubham.agrawal@harness.io",
        "username": "shubham149"
      },
      "committer": {
        "name": "GitHub",
        "email": "noreply@github.com",
        "username": "web-flow"
      },
      "added": [
  
      ],
      "removed": [
  
      ],
      "modified": [
        "README.md"
      ]
    }
}