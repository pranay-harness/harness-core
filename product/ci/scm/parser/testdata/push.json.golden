{
    "push": {
        "ref": "refs/heads/master",
        "repo": {
            "id": "288719292",
            "namespace": "wings-software",
            "name": "test-webhook",
            "branch": "master",
            "private": true,
            "clone": "https://github.com/wings-software/test-webhook.git",
            "cloneSsh": "git@github.com:wings-software/test-webhook.git",
            "link": "https://github.com/wings-software/test-webhook",
            "created": "0001-01-01T00:00:00Z",
            "updated": "0001-01-01T00:00:00Z"
        },
        "before": "e286f86f4db84e682b50007772a28a9d5e9b51d0",
        "after": "3c377597b64e441cae5378b660578fa00983aaba",
        "commit": {
            "sha": "3c377597b64e441cae5378b660578fa00983aaba",
            "message": "Update README.md",
            "author": {
                "name": "Shubham Agrawal",
                "email": "shubham.agrawal@harness.io",
                "date": "2020-08-25T07:30:01Z",
                "login": "shubham149"
            },
            "committer": {
                "name": "GitHub",
                "email": "noreply@github.com",
                "date": "2020-08-25T07:30:01Z",
                "login": "web-flow"
            },
            "link": "https://github.com/wings-software/test-webhook/compare/e286f86f4db8...3c377597b64e"
        },
        "sender": {
            "login": "shubham149",
            "avatar": "https://avatars1.githubusercontent.com/u/61136843?v=4",
            "created": "0001-01-01T00:00:00Z",
            "updated": "0001-01-01T00:00:00Z"
        },
        "commits": [
            {
                "sha": "3c377597b64e441cae5378b660578fa00983aaba",
                "message": "Update README.md",
                "author": {
                    "name": "Shubham Agrawal",
                    "email": "shubham.agrawal@harness.io",
                    "date": "2020-08-25T07:30:01Z",
                    "login": "shubham149"
                },
                "committer": {
                    "name": "GitHub",
                    "email": "noreply@github.com",
                    "date": "2020-08-25T07:30:01Z",
                    "login": "web-flow"
                },
                "link": "https://github.com/wings-software/test-webhook/commit/3c377597b64e441cae5378b660578fa00983aaba"
            }
        ]
    }
}